src/Symfony/Component/Form/Tests/Fixtures/FormWithSameParentTypeAsObject.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Fixtures;
+
+use Symfony\Component\Form\AbstractType;
+
+class FormWithSameParentTypeAsObject extends AbstractType
+{
+    public function getName()
+    {
+        return 'foo_type';
+    }
+
+    public function getParent()
+    {
+        return new FormWithSameParentTypeAsObject();
```

Comment:
Please, use `return new self();` to make fabbot happy again ;)

Reply:
I did consider changing this initially when I saw the failure, but I'm just concerned it would make the test less intuitive

Reply:
Why would it look less intuitive to you? Would be even more explicit to me :)

Could you do the change?